src/signatureUtils.js

```diff
@@ -0,0 +1,253 @@
+ * publicKeyUrl2 : Value of public key url retrieved from the x-adobe-public-key2-url header in each POST request to webhook
+ * @param {*} recipientClientId - target recipient client id
+ * @param {*} signedPayload - I/O Events proper signed payload
+ * @returns {boolean} true if either signatures are valid or false 
+ */
+async function verifyDigitalSignature(signatureOptions, recipientClientId, signedPayload) {
+    var signatures = [signatureOptions.digiSignature1, signatureOptions.digiSignature2]
+    var keys = await fetchPemEncodedPublicKeys(signatureOptions.publicKeyUrl1, signatureOptions.publicKeyUrl2)
+    return await verifySignature(signatures, signedPayload, keys, recipientClientId)
+}
+
+/**
+ * Feteched the pem encoded public keys either from the state lib cache or directly via cloud front url
+ * 
+ * @param {*} pubKeyUrl1 cloud front url of format https://d2wbnl47xxxxx.cloudfront.net/pub-key-1.pem
+ * @param {*} pubKeyUrl2 cloud front url of format https://d2wbnl47xxxxx.cloudfront.net/pub-key-1.pem
+ * @returns {array} of two public keys
+ */
+async function fetchPemEncodedPublicKeys(pubKeyUrl1, pubKeyUrl2) {
+    var pubKey1Pem, pubKey2Pem
+    try {
+        const state = await stateLib.init()
+        pubKey1Pem = await fetchPubKeyFromCacheOrApi(pubKeyUrl1, state)
+        pubKey2Pem = await fetchPubKeyFromCacheOrApi(pubKeyUrl2, state)
+    } catch (error) {
+        logger.error('error occurred while fetching pem encoded public keys either from cache or public key urls due to %s', error.message)
+        return genErrorResponse(500, 'Error occurred while fetching pem encoded Public Key')
+    }
+    return [pubKey1Pem, pubKey2Pem]
+}
+
+/**
+ * Wrapper to fetch the key from aio-lib-state, if not present fetch using
+ * the cloud front url and set in aio-lib-state 
+ * 
+ * @param {*} pubKeyUrl cloud front url of format https://d2wbnl47xxxxx.cloudfront.net/pub-key-1.pem
+ * @param {*} state aio-lib-state client
+ * @returns public key
+ */
+async function fetchPubKeyFromCacheOrApi(pubKeyUrl, state) {
+
+    const publicKeyFileName = await getPubKeyFileName(pubKeyUrl)
+    let publicKey = await getKeyFromCache(state, publicKeyFileName)
+    if (publicKey) {
+        return publicKey
+    }
+    logger.info('public key %s not present in aio state lib cache, fetching directly from the url..', publicKeyFileName)
+    // fetch from api and set in cache default expiry 24h
+    publicKey = await fetchPublicKeyFromCloudFront(pubKeyUrl)
+    await state.put(publicKeyFileName, publicKey)
+    return publicKey
+}
+
+/**
+ * Gets the public key from cache
+ * 
+ * @param {*} state aio state lib instance
+ * @param {*} publicKeyFileNameAsKey public key file name in format pub-key-1.pem
+ * @returns public key
+ */
+async function getKeyFromCache(state, publicKeyFileNameAsKey) {
+    return await state.get(publicKeyFileNameAsKey)
+}
```

Comment:
Do we really need a function for this?

Reply:
I segregated it include it for test coverage.